src/main/java/ama/post/PostController.java

```diff
@@ -64,7 +69,36 @@ public String addCommentToSubmission(@PathVariable(value = "submission") String
             return "redirect:/posts/{submission}";
         }
         return "pageNotFound";
+    }
 
+    @GetMapping("/comments/upvote")
+    public String upvoteComment(HttpServletRequest request, @RequestParam("id") String commentId) {
+        CommentPost comment = commentPostRepository.findById(Long.parseLong(commentId)).get(0);
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        String username = auth.getName();
+        User user = userRepository.findByUsername(username);
+        if (user != null && comment != null) {
+            comment.upvote(user);
+            commentPostRepository.save(comment);
+            String referer = request.getHeader("Referer");
```

Comment:
Referer is the URL of the user when this request was made. This allows us to redirect back to this ama's page without having a reference to the submission.